pypeit/coadd2d.py

```diff
@@ -501,25 +508,48 @@ def reduce(self, pseudo_dict, show=None, show_peaks=None, basename=None):
         slits = self.stack_dict['slits_list'][0]
         reduce_bpm = (slits.mask > 0) & (np.invert(slits.bitmask.flagged(
             slits.mask, flag=slits.bitmask.exclude_for_reducing)))
-        redux.reduce_bpm = reduce_bpm
+        objFind.reduce_bpm = reduce_bpm
 
         if show:
             gpm = sciImage.select_flag(invert=True)
-            redux.show('image', image=pseudo_dict['imgminsky']*gpm.astype(float),
+            objFind.show('image', image=pseudo_dict['imgminsky']*gpm.astype(float),
                        chname='imgminsky', slits=True, clear=True)
 
         # TODO:
         #  Object finding, this appears inevitable for the moment, since we need to be able to call find_objects
         #  outside of reduce. I think the solution here is to create a method in reduce for that performs the modified
         #  2d coadd reduce
-        sobjs_obj, nobj, skymask_init = redux.find_objects(
-            sciImage.image, show_peaks=show_peaks, save_objfindQA=True,
-            manual_extract_dict=manual_dict)
+        sobjs_obj, nobj = objFind.find_objects(sciImage.image, show_peaks=show_peaks,
+                                               save_objfindQA=True, manual_extract_dict=manual_dict)
+
+        # Initiate Extract object
+        extract = extraction.Extract.get_instance(sciImage, sobjs_obj, self.spectrograph, parcopy, caliBrate,
```

Comment:
Can we name this exTract or something that is not the same name as core.extract. There is no collision as of now, but the call extract.local_skysub_extract could be confused with the same call from core.extract.